### ISSUE TITLE
ListItem not firing onPress on some devices

### ISSUE THREAD

**Issue by lesmo** [akveo/react-native-ui-kitten#874]
## 🐛 Bug Report

`ListItem` won't trigger `onPress` on some devices, the callback is not called on iPhone 5s, iPhone SE and Samsung S8+ so far (but I fear more devices are affected). However, it's correctly called every time on simulators for iPhone 11, iPhone 8, and iPhone SE, physical Xiaomi Redmi Note 8 Pro.

## To Reproduce

I'm using the `ListItem` as a sort of button on it's own:

```js
<ListItem
  title={intl.formatMessage(messages.about)}
  onPress={() => navigation.navigate('userAbout')}
/>
```

But the same problem occurs on a `List` I'm using:
```js
<>
  <List
    data={filteredData}
    initialNumToRender={15}
    maxToRenderPerBatch={7}
    windowSize={3}
    renderItem={({ item }) => (
      <ListItem
        key={item.id}
        title={item.title}
        onPress={() =>
          // This is not called on some devices
          navigation.navigate('someRoute', {
            id: item.id,
          })
        }
      />
    )}
  />
</>
```

At first I thought it could be [related to some positioning bug](https://github.com/facebook/react-native/issues/22397) but, since it's happening on iOS and Android, and in the case of it being used inside a `List` with no weird positioning, none of the related issues apply.

## Expected behavior

`onPress` should be called successfully on every device, every time.

## Link to runnable example or repository (highly encouraged)

Hopefully soon I'll get a repro-repo.

## UI Kitten and Eva version

| Package      | Version |
| ----------- | ----------- |
| @eva-design/eva      | 1.3.1     |
| @ui-kitten/components   | 4.4.0-beta.1        |

## Environment information

```bash
  System:
    OS: macOS Mojave 10.14.6
    CPU: (4) x64 Intel(R) Core(TM) i7-3520M CPU @ 2.90GHz
  Binaries:
    Node: 10.16.0 - ~/.nvm/versions/node/v10.16.0/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.13.0 - ~/.nvm/versions/node/v10.16.0/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
    Android SDK:
      API Levels: 22, 23, 26, 27, 28
      Build Tools: 28.0.2, 28.0.3, 29.0.2
      System Images: android-24 | Google Play Intel x86 Atom
  IDEs:
    Android Studio: 3.1 AI-173.4907809
    Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.5 => 0.61.5
```

**Comment by lesmo** [comment]
I've also written custom components using `TouchableOpacity` (the underlying component used by `ListItem`) but they don't seem affected by any changes I've made. Only specifically `ListItem`, including a custom "fork" I made where I copied `ListItem` and modified it to accept any component instead of just icons...

This is pretty strange.

**Comment by lesmo** [comment]
Alright, further investigation seems to indicate this bug is more closely related to React Native itself.

I'm guessing that when the JS code's FPS drops a bit, on some devices React Native won't recognize an `onPress` event on _"heavy"_ components. One of my _"heavy"_ components (several components, styling, etc.) still handled `onPress` timely and as expected. It _feels_ related to `ScrollView` (which is used by the underlying `FlatList` component), but I think it needs further investigation (that I won't make for now, and is obviously out of scope for `ui-kitten`).

**My solution**: I was able to refactor some code to use `onPressOut` and `onPressIn` where applicable. For my `List`, I ended up listening to both `onPress` and `onLongPress` which seems to work across devices as expected. I'm rolling out this fix as I'm writing this.

Thus... I'm closing this issue, as it's apparently not directly an issue of `ui-kitten`.

**Comment by artyorsh** [comment]
@lesmo thanks for the explanation! Could I ask you to post a better solution if you find it? :)

**Comment by lesmo** [comment]
@artyorsh sure! But so far I've figured out it has more to do with JS thread FPS dropping to almost 0 for some operations in the background, thus the UI won't properly react to user input. I've delayed many things with `InteractionManager` so UI won't be affected, and so far that has worked pretty well - although I haven't tried to "rollback" my weird dual `onPress` + `onLongPress` on `ListItem` yet.

I think another solution would be to use `requestAnimationFrame` on the call... I'll need to make a repro-repo to try out these theories.
